File: Data Structures & Algorithms/balanced-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def isBalanced(self, root: Optional[TreeNode]) -> bool:

        def dfs(node):
            if node == None:
                return 0

            left = dfs(node.left)
            right = dfs(node.right)
            
            if abs(left-right) > 1:
                raise Exception("unbalanced")

            return 1 + max(left, right)


        try:


            dfs(root)
            return True
        except:
            return False

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.java
class Solution {
    public int maxProfit(int[] prices) {
        int left = 0;
        int right = 1;
        int max = 0;
        if (prices.length == 1) {
            return 0;
        }
        

        while (right < prices.length) {
            if(prices[left] >= prices[right]) {
                left = right;
            }

            else {
                if (prices[right] - prices[left] > max) {
                    max = prices[right] - prices[left];
                }
            }

            right += 1;
        }   

        return max;

    }
}
